x/supplier/keeper/claim.go

```diff
@@ -9,19 +9,20 @@ import (
 	"github.com/pokt-network/poktroll/x/supplier/types"
 )
 
-// InsertClaim adds a claim to the store
-func (k Keeper) InsertClaim(ctx sdk.Context, claim types.Claim) {
-	logger := k.Logger(ctx).With("method", "InsertClaim")
+// UpsertClaim inserts or updates a specific claim in the store by index.
+func (k Keeper) UpsertClaim(ctx sdk.Context, claim types.Claim) {
```

Comment:
Upsert is a new term for me ahah I like it

Reply:
For my own understanding how could a claim be updated? Surely there is a single claim per supplier per session - and this should not be updateable no?

Reply:
This is a good point. 

If the supplier made a mistake and submitted the wrong claim, we want to give them an opportunity to update the claim and still receive reward for the work in the session. 

However, this introduces logic on when do we start blocking update operations (e.g. after the claim submission window has passed).

@bryanchriswhite: I was going to suggest changing this back from `Upsert` to `Insert`, but after thinking about it, could you do the following instead: add `func (k msgServer) CreateClaim`:

```go
TODO_BLOCKER: Prevent Claim upserts after the ClaimWindow is closed.
TODO_BLOCKER: Validate the signature on the Claim message corresponds to the supplier before Upserting
```

Reply:
That's a great question! 😅🤦